### PR TITLE
#6Task. ivan.lozovskij

### PR DESCRIPTION
**Pull request by Lozovskij**
https://travis-ci.org/github/Lozovskij/js-assignments/builds/705717535

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -209,7 +245,10 @@ function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
  * 'noon' => 'noon'
  */
 function reverseString(str) {
-    throw new Error('Not implemented');
+    var splitString = str.split("");
+    var reverseArray = splitString.reverse(); 
+    var joinArray = reverseArray.join("");
+    return joinArray;
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can write the same in 1 row :)

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -226,7 +265,7 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    return parseInt(reverseString(num + ''));
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can't use previous function, sorry

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -270,8 +321,23 @@ function isCreditCardNumber(ccn) {
...
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    var res = 0;
+    res = getSum(num);
+    while (res >= 10) {
+        res = getSum(res);
+    }
+    return res;
 }
+function getSum(num) {
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can't create function outside the main function, sorry
And this solution can be done in 3 rows :)

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -296,7 +362,25 @@ function getDigitalRoot(num) {
...
+    return stack.length == 0;
+}
+function isCloser(ch) {
+    return ")]}>".indexOf(ch) != -1;
+}
+function getOpener(ch) {
+    var openers = "<[{(";
+    var closers = ">]})";
+    return openers[closers.indexOf(ch)];
 }
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Same. You can't create functions outside the main function, sorry

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -332,7 +416,32 @@ function isBracketsBalanced(str) {
...
+    else if (hoursLag > 36 && daysLag <= 25) {return `${getNumber(daysLag)} days ago`;}
+    else if (hoursLag > 22 && hoursLag <= 36) {return "a day ago";}
+    else if (minutesLag > 90 && hoursLag <= 22) {return `${getNumber(hoursLag)} hours ago`;}
+    else if (minutesLag > 45 && minutesLag <= 90) {return "an hour ago";}
+    else if (secondsLag > 90 && minutesLag <= 45) {return `${getNumber(minutesLag)} minutes ago`;}
+    else if (secondsLag > 45 && secondsLag <= 90) {return "a minute ago";}
+    else {return "a few seconds ago";}
+}
+
+function getNumber(i) {
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Same

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -373,7 +482,19 @@ function toNaryString(num, n) {
  *   ['/web/favicon.ico', '/web-scripts/dump', '/webalizer/logs'] => '/'
  */
 function getCommonDirectoryPath(pathes) {
-    throw new Error('Not implemented');
+    return trimToFolderSign(getLongestCommonPrefix(pathes));
+}
+function getLongestCommonPrefix(pathes) {
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: Same

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -251,7 +290,19 @@ function reverseInteger(num) {
  *   4916123456789012 => false
  */
 function isCreditCardNumber(ccn) {
-    throw new Error('Not implemented');
+    let sum = 0;
+    var digits = ccn + "";
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: It's better to use toString() method instead of ccn + ""
This will save 3 characters.

**Comment by elizavetaShcherbatykhAis** [comment]
Please check the comments.
I'll mark the PR as draft please click "ready for review" when it will be corrected!
Thank you!

**Comment by Lozovskij** [comment]
https://travis-ci.org/github/Lozovskij/js-assignments/builds/710120585

File: task/06-conditions-n-loops-tasks.js [hunk 8]
```diff
@@ -226,7 +262,13 @@ function reverseString(str) {
...
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    let res = 0,i = num;
+
+    while (i) {
+        res = res * 10 + i % 10;
+        i = Math.floor(i / 10);
+    }
+    return res;
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js]: You can do it in 1 row also :)

File: task/06-conditions-n-loops-tasks.js [hunk 9]
```diff
@@ -270,10 +324,14 @@ function isCreditCardNumber(ccn) {
...
  */
 function getDigitalRoot(num) {
-    throw new Error('Not implemented');
+    const reducer = (accumulator, currentValue) => accumulator + parseInt(currentValue);
+    let sum = num.toString().split("").reduce(reducer, 0);
+    if (sum <= 9) {
+        return sum;
+    }
+    return getDigitalRoot(sum);
 }
```
Comment (elizavetaShcherbatykhAis) [inline task/06-conditions-n-loops-tasks.js:327]: Super!

**Comment by elizavetaShcherbatykhAis** [comment]
Please check the comments.
You in one step of accepted PR :)
I'll mark the PR as draft please click "ready for review" when it will be corrected!
Thank you!

**Comment by Lozovskij** [comment]
https://travis-ci.org/github/Lozovskij/js-assignments/builds/710309213

**Comment by elizavetaShcherbatykhAis** [comment]
Well done! I close PR as accepted

**Pull request closed without merging**